### ISSUE TITLE
[BUG] deepspeed_stage_3 was used in pytorch_lightning。when initialize, it cost huge  cpu memory  which increase with the grow of gpu num。

### ISSUE THREAD

**Issue by linyubupa** [microsoft/DeepSpeed#2997]
**Describe the bug**
deepspeed_stage_3 was used in pytorch_lightning。when initialize, it cost huge  cpu memory 。

the amount of cpu memory used = gpu_number * 2 * model_size 

**To Reproduce**

**Expected behavior**

**ds_report output**
Please run `ds_report` to give us details about your setup.

**Screenshots**

**System info (please complete the following information):**
 - OS: [e.g. Ubuntu 18.04]
 - GPU count and types [e.g. two machines with x8 A100s each]
 - Interconnects (if applicable) [e.g., two machines connected with 100 Gbps IB]
 - Python version
 - Any other relevant info about your setup

**Launcher context**
Are you launching your experiment with the `deepspeed` launcher, MPI, or something else?

**Docker context**
Are you using a specific docker image that you can share?

**Additional context**

**Comment by linyubupa** [comment]
the pytorch-lightning code was used  : 

trainer = Trainer(
        max_epochs=1,
        devices=args.num_devices,
        precision=16,
        strategy="deepspeed_stage_3",
        accelerator='gpu',
        num_nodes=args.num_nodes,
        limit_val_batches=0,
        # 添加plugins
        plugins=plugins,
        # 添加log和profile
        logger=lighting_logger,
        profiler=profiler,
        # 添加callback
        callbacks=callbacks,
        # 关闭官方进度条
        enable_progress_bar=False
    )

    logger.info("start train #########################")
    trainer.fit(model, data_module)

**Comment by linyubupa** [comment]
the amount of cpu memory used = gpu_number * 2 * model_size

**Comment by ShijieZZZZ** [comment]
Hi @linyubupa, could you describe more details about reproducing this issue? Especially how you measured _cpu memory used_ and _model_size_

**Comment by linyubupa** [comment]
> Hi @linyubupa, could you describe more details about reproducing this issue? Especially how you measured _cpu memory used_ and _model_size_

this is the code that i used:

**Comment by linyubupa** [comment]
i think the most cause of this result is 
mp.spawn(main_worker, args=(sys.argv,), nprocs=gpu_count, join=True)

**Comment by linyubupa** [comment]
and this is the config of ds

{
  "train_batch_size": "auto",
  "fp16": {
    "enabled": true,
    "min_loss_scale": 1,
    "opt_level": "auto"
  },
  "zero_optimization": {
    "stage": 3,
    "allgather_partitions": true,
    "allgather_bucket_size": 5e8,
    "contiguous_gradients": true
  },
  "optimizer": {
    "type": "AdamW",
    "params": {
      "lr": 1e-05,
      "betas": [
        0.9,
        0.95
      ],
      "eps": 1e-08
    }
  },
  "scheduler": {
    "type": "WarmupLR",
    "params": {
      "warmup_min_lr": 0,
      "warmup_max_lr": 1e-05,
      "warmup_num_steps": "auto"
    }
  }
}

**Comment by linyubupa** [comment]
> Hi @linyubupa, could you describe more details about reproducing this issue? Especially how you measured _cpu memory used_ and _model_size_

I measured cpu memory by using aistudio tools which is proved correct
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/20812813/226087601-a04e6e1f-31c4-44f1-96ed-4e7ffbdc91a7.png">

**Comment by ShijieZZZZ** [comment]
Hi @linyubupa, I am not able to run your code. Do you have a smaller test (but complete setup) for reproducing?
Your suspicion on `mp.spawn` makes sense. See this https://github.com/pytorch/pytorch/issues/38645.
Could you also try measureing cpu usage with `deepspeed.runtime.utils.see_memory_usage`?

**Comment by ShijieZZZZ** [comment]
Hi @linyubupa. Closed for now. Feel free to reopen if there is any update.
